### DISCUSSION TITLE
Missing TyreLife and wrong Stint

### DISCUSSION THREAD

**Discussion by harish5p** [theOehrly/Fast-F1#197] [General]
For Canadian GP race session, when we select the fastest laps for each driver. In the `TyreLife` column we have NaNs for 'RUS' and 'NOR'. A suggestion would be to fill NaNs assuming TyreLife as 1 for first lap after pit stop

When I looked into 'NOR' laps data, on `LapNumber` 43, he is on `Stint` 3 on `Compound` HARD, but for next lap(LapNumber=44), he is still on `Stint` 3 but on `Compound` MEDIUM.  [Lando had a pit stop on Lap 42 according to the official data here](https://inmotion.dhl/en/formula-1/fastest-pit-stop-award) I'm quite confused.

**Comment by theOehrly** [comment]
I finally got around to investigating this. Can confirm that this is in fact a bug in the API parser code. The bug is causing the NaN values as well as the incorrect compound shown. No ETA though for when I'll have a fix for it. I'm rather busy at the moment.

Just in general, I will not automatically fill in nan values with an assumption for tyre life. The api should remember tyre age over multiple sessions. Therefore, it will often be wrong to assume TyreLife = 1 at the beginning of a stint.
